### ISSUE TITLE
Using actuals values instead of forecast values in validation data

### ISSUE THREAD

**Issue by AdrianAntico** [nredell/forecastML#19]
Are your functions utilizing actual values for the lags and rolling stats in the validation data set you utilize for gauging accuracy or are you generating one-step-ahead forecasts then creating the lags and rolling stats off of the predicted values, and repeating that process throughout the number of validation periods?

**Comment by nredell** [comment]
Fair question. There's no rolling or iterated anything in this package. All actuals. To use an example: If you're training a 1- to 3-month out forecasting model using 2010 to 2019 monthly actuals and 2017 happened to be a particularly tough year to forecast, you could hold out all of 2017 and see which models fair best at forecasting here. So the holdout window doesn't have to be 3 months in 2017 in this case. The 2017 holdout window is vanilla nested cross-validation where your model is learning a typical ML mapping from Xs to Y and is only privy to things that happened in 2010:2016 and 2018:2019. The error/accuracy metrics would then reflect an exact 3-month horizon error, not a 1- to 3-month error/accuracy rate...although a 3-month model would generate forecasts for 1- to 3-steps ahead.

I'm open to suggestions here because I could see wanting 1-step-ahead performance without having to train a dedicated 1-step-ahead model.

A tangential concern is that, for example, data from the training dataset from 2016 and 2018 is leaking into the 2017 holdout window/data when lagged features are used...which is why a number of special time series cross-validation measures have been developed. The punchline of the paper that I'm basing this package on is that as long as there is no residual correlation in the fitted models, this leaking of info is a non-problem.

**Comment by AdrianAntico** [comment]
Thanks for your response. Shoot me an email if you want to discuss this a bit more. I have quite a bit of ideas... adrianantico@gmail.com

**Comment by nredell** [comment]
Note to self: With version 0.8.0 now on CRAN with a fairly substantial refactor, I'm to the point where adding a variety of error metrics makes sense. Closing now but will re-open when I intend to work on it.

**Comment by AdrianAntico** [comment]
Thanks for the update - by the way, check out the new models I just released to my package. I enhanced my CARMA functions (currently just AutoCatBoostCARMA()) which is an ML based forecasting system. Tons of features on it. I also just released a model called AutoBanditSarima()
